src/account/UTXOTxParameters.cpp

```diff
@@ -0,0 +1,45 @@
+#include "UTXOTxParameters.hpp"
+#include <boost/multiprecision/fwd.hpp>
+
+sgns::UTXOTxParameters::UTXOTxParameters( const std::vector<GeniusUTXO>          &utxo_pool,
+                                          const KeyGenerator::ElGamal::PublicKey &src_address,
+                                          const std::vector<OutputDestInfo>      &destinations,
+                                          std::string                             signature )
+{
+    int64_t total_amount = 0;
+
+    for ( auto &dest_info : destinations )
+    {
+        total_amount += static_cast<int64_t>( dest_info.encrypted_amount );
+    }
+
+    int64_t remain = total_amount;
+
+    for ( auto &utxo : utxo_pool )
+    {
+        if ( remain <= 0 )
+        {
+            break;
+        }
+        InputUTXOInfo curr_input{ utxo.GetTxID(), utxo.GetOutputIdx(), std::move( signature ) };
```

Comment:
If you std::move this it will lose the value the first iteration.

Reply:
🤡